azavea/pfb-network-connectivity
# Feature/awf/analysis mount tmp to ephemeral

## Overview

Updates the analysis job configuration to mount `/tmp` to the NVME scratch space drives on AWS.

Leaving /tmp within the containers leads to storage issues since larger jobs use many GB of files which rapidly eats into the available space on an instance's docker drive pool.

Instead we mount /tmp to a volume attached to the host scratch space, which has plenty of room for our tmp files. 

### Demo

Note /tmp file paths in screenshot, now contain job id and subdirectories:
![screen shot 2017-04-27 at 13 35 46](https://cloud.githubusercontent.com/assets/1818302/25496332/8989aa80-2b4e-11e7-914c-a409e80334b6.png)

![screen shot 2017-04-27 at 13 43 27](https://cloud.githubusercontent.com/assets/1818302/25496612/885f2012-2b4f-11e7-9915-902a3fecfaa5.png)

Fix for #359:
![screen shot 2017-04-27 at 13 44 18](https://cloud.githubusercontent.com/assets/1818302/25496676/cd60d548-2b4f-11e7-9dc0-b04dea9bea7f.png)

Staging disk usage not exploding when running LA:
![screen shot 2017-04-27 at 14 03 15](https://cloud.githubusercontent.com/assets/1818302/25497364/50b3532e-2b52-11e7-8037-03b89fa01b89.png)

### Notes

I originally only mounted `/tmp` leaving all other file paths in the analysis jobs the same. But it seems dangerous to have potentially multiple jobs mounting and using the same available space. So, I updated analysis to be able to take a `NB_TEMPDIR` option which specifies the temp directory to use for all transient file operations. This allows the Django application to properly namespace the directories mounted to the job containers. The alternative would be to modify the volume/mountpoint paths when we call Batch's SubmitJob, but there is no option to do that, those settings can only be modified in JobDefinition.

## Testing Instructions

If desired, can test changes locally by setting `NB_TEMPDIR="/tmp/${PFB_JOB_ID}"` in `./scripts/run-local-analysis` and then passing in `NB_TEMPDIR` by adding an extra env line in the docker run command: `-e NB_TEMPDIR=${NB_TEMPDIR}` before triggering a local job. You should see the tmp dir created and the printed file paths change appropriately. 

Alternatively, staging LA job has now progressed past where the production one failed due to filling up the root volume (see above screenshot).

Closes #358 #359
